File: projects/05/quadratic_formula.rb
# quadratic_formula.rb computes quadratic formula given a b c values
#
# 3/5/18
# Bryce Allen
# cs 214
# Calvin College

root1 = 0.0
root2 = 0.0
def quadratic_formula(a, b, c)
  if a == 0
    print "invalid input"
  elsif ((b ** 2) - (4 * a * c)) < 0
    print "invalid input"
  else
    print "The roots are: "
    root1 = ((-b + Math.sqrt(((b ** 2) - (4 * a * c))))/(2 * a))
    print root1
    print " and "
    root2 = ((-b - Math.sqrt(((b ** 2) - (4 * a * c))))/(2 * a))
    print root2
    print " "
  end
end


if __FILE__ == $0
  print "enter a b and c:"
  a = gets
  b = gets
  c = gets
  quadratic_formula(a.to_i, b.to_i, c.to_i)
end

File: projects/05/QuadraticFormula.java
/* QuadraticFormula computes the roots of a quadratic expression
 * cs 214 Calvin College
 * Completed by: Bryce Allen
 * 3/2/18
 *
 */
import java.util.Scanner;

public class QuadraticFormula
{
    private static double root1;
    private static double root2;
    
    public static void main(String[] args)
    {
	Scanner keyboard = new Scanner(System.in);
	
	System.out.println("Enter the A B and C values");
	double a = keyboard.nextDouble();
	double b = keyboard.nextDouble();
	double c = keyboard.nextDouble();

        root1 = 0.0;
        root2 = 0.0;

	if( quadraticFormula(a,b,c) )
	    {
		System.out.println("Root one " + root1);
		System.out.println("Root two " + root2);
	    }
	else{ System.out.println("Not valid input"); }
    }

    public static boolean quadraticFormula(double a, double b, double c)
    {
	if( a == 0 )
	    return false;
	
	double test = (Math.pow(b, 2.0)-(4*a*c));
	if( test < 0 )
	    return false;

	root1 = ((-b + (Math.sqrt(test)))/(2*a));
	root2 = ((-b - (Math.sqrt(test)))/(2*a));

	return true;
    }
}
		       
